registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -1020,7 +1024,13 @@ private IdResponseDTO linkRegIdWrtUin(String lostPacketRegId, String matchedRegI
 			identityObject.put(UINConstants.UIN, uin);
 			String schemaVersion = packetManagerService.getFieldByMappingJsonKey(lostPacketRegId, MappingJsonConstants.IDSCHEMA_VERSION, process, ProviderStageName.UIN_GENERATOR);
 			identityObject.put(idschemaversion, convertIdschemaToDouble ? Double.valueOf(schemaVersion) : schemaVersion);
-
+			String[] upd= updateInfo.split(",");
```

Comment:
make default value for updateInfo as null. If no update is allowed then skip this code
also rename the method name -
linkRegIdWrtUin => lostAndUpdateUin()

Reply:
done